core/src/main/java/com/sequenceiq/cloudbreak/job/nodestatus/NodeStatusCheckerJob.java

```diff
@@ -97,43 +110,87 @@ private void executeNodeStatusCheck(Stack stack) {
             }
             CdpNodeStatusRequest request = requestBuilder.withCmMonitoring(true).build();
             CdpNodeStatuses statuses = nodeStatusService.getNodeStatuses(stack, request);
-            processNodeStatusReport(statuses.getNetworkReport(), NodeStatusProto.NodeStatus::getNetworkDetails, stack, "Network");
-            processNodeStatusReport(statuses.getServicesReport(), NodeStatusProto.NodeStatus::getServicesDetails, stack, "Services");
-            processNodeStatusReport(statuses.getSystemMetricsReport(), NodeStatusProto.NodeStatus::getSystemMetrics, stack, "System metrics");
-            processCmMetricsReport(statuses.getCmMetricsReport(), stack);
+            processNodeStatusReport(statuses.getNetworkReport(), NodeStatusProto.NodeStatus::getNetworkDetails, stack, "Network", accountId);
+            processNodeStatusReport(statuses.getServicesReport(), NodeStatusProto.NodeStatus::getServicesDetails, stack, "Services", accountId);
+            final boolean processMetrics;
             if (StackType.WORKLOAD.equals(stack.getType())) {
-                processNodeStatusReport(statuses.getMeteringReport(), NodeStatusProto.NodeStatus::getSystemMetrics, stack, "Metering");
+                processMetrics = entitlementService.datahubMetricsDatabusProcessing(accountId);
+                processNodeStatusReport(statuses.getMeteringReport(), NodeStatusProto.NodeStatus::getSystemMetrics, stack, "Metering", accountId);
+            } else {
+                processMetrics = entitlementService.datalakeMetricsDatabusProcessing(accountId);
             }
+            processNodeStatusReport(statuses.getSystemMetricsReport(), NodeStatusProto.NodeStatus::getSystemMetrics,
+                    stack, "System metrics", accountId, processMetrics);
+            processCmMetricsReport(statuses.getCmMetricsReport(), stack, accountId, processMetrics);
         }
     }
 
-    private void processCmMetricsReport(Optional<RPCResponse<NodeStatusProto.CmMetricsReport>> cmMetricsReport, Stack stack) {
+    private void processCmMetricsReport(Optional<RPCResponse<NodeStatusProto.CmMetricsReport>> cmMetricsReport, Stack stack,
+            String accountId, boolean process) {
         if (cmMetricsReport.isPresent()) {
             LOGGER.debug("CM metrics report: {}", cmMetricsReport.get().getFirstTextMessage());
```

Comment:
is this log statement that important? Otherwise, you could get rid of the boolean process argument and just not call this method based on the boolean value.

Reply:
yes, so if databus processing is not enabled, which can be the case, it will logs out the response on CB side, which is still useful, also it's possible we are just getting json output but not fitting with the nodestatus proto (in earlier nodestatus monitor changes, there were no proto objects implemented ... there was a different json based response ... so in that case, proto object is not filled but we still have a text response that can be logged out)